### PR TITLE
Casing and other minor typo fixes in docs

### PR DESCRIPTION
**Pull request by mingaldrichgan**
Pretty self-explanatory CR; let me know if you need more details.

### CODE REVIEW

File: docs/docs/typography-js.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: Typography.js
 ---
 
-# Using Typography.js in Gatsby
+## Using Typography.js in Gatsby
```
Comment (mingaldrichgan) [inline docs/docs/typography-js.md]: Was unsure about this; the [style guide](https://www.gatsbyjs.org/docs/gatsby-style-guide/#format-code-blocks-inline-code-and-images) says to avoid H1. I left the other headers unchanged. Happy to undo this change or decrement the other headers accordingly, as appropriate. Let me know!

File: docs/docs/how-code-splitting-works.md [hunk 2]
```diff
@@ -12,9 +12,8 @@ During the [Write Out Pages](/docs/write-pages/#async-requiresjs) bootstrap phas
 
 ```javascript
 exports.components = {
-  "component--src-blog-js": () =>
-    import("/home/site/src/blog.js"),
-    /* webpackChunkName: "component---src-blog-js" */
+  "component--src-blog-js": () => import("/home/site/src/blog.js"),
+  /* webpackChunkName: "component---src-blog-js" */
```
Comment (m-allanson) [inline docs/docs/how-code-splitting-works.md]: I think this is a pre-existing typo, the `webpackChunkName` comment should be part of the import call:

```diff
-  "component--src-blog-js": () => import("/home/site/src/blog.js"),
-   /* webpackChunkName: "component---src-blog-js" */
+  "component--src-blog-js": () => import("/home/site/src/blog.js" /* webpackChunkName: "component---src-blog-js" */),
```

File: docs/docs/typography-js.md (hunk 1 above)
Comment (m-allanson) [reply docs/docs/typography-js.md]: 👍 the `title: Typography.js` block will be set as an h1, so bumping this to h2 is perfect.

**Comment by mingaldrichgan** [comment]
@m-allanson Thanks, fixed.
